.github/workflows/build.yml

```diff
@@ -40,6 +40,22 @@ jobs:
     - run: ./canvas_modules/harness/build.sh test
     - run: ./canvas_modules/harness/functional_test.sh
 
+  sonarcloud:
+    if: github.ref == 'refs/heads/main'
+    if: github.repository == 'elyra-ai/canvas'
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        node-version: 20.x
+    - name: Install common-canvas dependencies
+      run: npm install --prefix canvas_modules/common-canvas
+    - name: SonarCloud Scan
```

Comment:
Can we not just move this to line 42?  Only the SonarCloud Scan step and add a conditional check on that step.